client/ayon_core/hosts/nuke/api/utils.py

```diff
@@ -142,3 +149,174 @@ def is_headless():
+
+    Args:
+        context (dict): Pyblish context.
+
+    Returns:
+        tuple: A tuple containing a boolean indicating success and a string
+        representing the error message.
+    """
+
+    error_message = ""
+    success = True
+    for result in context.data["results"]:
+        if result["success"]:
+            continue
+
+        success = False
+
+        err = result["error"]
+        formatted_traceback = "".join(
+            traceback.format_exception(
+                type(err),
+                err,
+                err.__traceback__
+            )
+        )
+        fname = result["plugin"].__module__
+        if 'File "<string>", line' in formatted_traceback:
+            _, lineno, func, msg = err.traceback
+            fname = os.path.abspath(fname)
+            formatted_traceback = formatted_traceback.replace(
+                'File "<string>", line',
+                'File "{0}", line'.format(fname)
+            )
+
+        err = result["error"]
+        error_message += "\n"
+        error_message += formatted_traceback
+
+    return success, error_message
+
+
+def submit_headless_farm(node):
+    # Ensure code is executed in root context.
+    if nuke.root() == nuke.thisNode():
+        _submit_headless_farm(node)
+    else:
+        # If not in root context, move to the root context and then execute the
+        # code.
+        with nuke.root():
+            _submit_headless_farm(node)
+
+
+def _submit_headless_farm(node):
+    """Headless farm submission
+
+    This function prepares the context for farm submission, validates it,
+    extracts relevant data, copies the current workfile to a timestamped copy,
+    and submits the job to the farm.
+
+    Args:
+        node (Node): The node for which the farm submission is being made.
+    """
+
+    context = util.collect()
+
+    success, error_report = create_error_report(context)
+
+    if not success:
+        show_message_dialog(
+            "Collection Errors", error_report, level="critical"
+        )
+        return
+
+    # Find instance for node and workfile.
+    instance = None
+    instance_workfile = None
+    for _instance in context:
+        if _instance.data["family"] == "workfile":
+            instance_workfile = _instance
+            _instance.data["active"] = False
+            continue
+
+        instance_node = _instance.data["transientData"]["node"]
+        if node.name() == instance_node.name():
+            instance = _instance
+        else:
+            _instance.data["active"] = False
+
+    if instance is None:
+        show_message_dialog(
+            "Collection Error",
+            "Could not find the instance from the node.",
+            level="critical"
+        )
+        return
+
+    # Enable for farm publishing.
+    instance.data["farm"] = True
+    instance.data["transfer"] = False
+
+    # Clear the families as we only want the main family, ei. no review etc.
+    instance.data["families"] = []
```

Comment:
This can cause a lot of issues.

Reply:
Have you got a solution in mind or maybe what issues it could cause?

Reply:
Nothing from this should happen out of publish plugins. So the question is how to affect publish logic from here.

You can create pyblish context object and set a specific key on it so some collector can do what are you doing here. So you can set `context.data["singleWriteSubmitNode"] = node`, take this as an example, can be done by instance id, node path etc. . Then add collector which in fact disables other instances and changes the attributes as necessary (can be order `pyblish.api.CollectorOrder - 0.4999`).

Also I think you should create `CreateContext` and pass it to the context too, to make sure the instance you're going to publish is enabled, because it won't be collected if is disabled.

Reply:
@iLLiCiTiT how do you pass the `CreateContext` to the publishing?
Tried the below but it does not work;
```
from pyblish import util

from ayon_core.pipeline.create import CreateContext
from ayon_core.pipeline import registered_host


host = registered_host()
create_context = CreateContext(host)
context = util.collect(create_context)
```

Reply:
```
import pyblish.api
from pyblish import util

from ayon_core.pipeline.create import CreateContext
from ayon_core.pipeline import registered_host


host = registered_host()
create_context = CreateContext(host)
publish_context = pyblish.api.Context()
publish_context.data["create_context"] = create_context
util.collect(context=publish_context)
```